packages/aws-cdk-lib/aws-iam/lib/principals.ts

```diff
@@ -264,6 +264,18 @@ export class PrincipalWithConditions extends PrincipalAdapter {
     this.additionalConditions = conditions;
   }
 
+  public addToAssumeRolePolicy(doc: PolicyDocument) {
+    // eslint-disable-next-line @typescript-eslint/no-require-imports
+    const adapter: typeof import('./private/policydoc-adapter') = require('./private/policydoc-adapter');
```

Comment:
Ok, I'll ask the question: why are we doing this weird import here? I assume something is stopping us from importing `MutatingPolicyDocumentAdapter` the normal way?

Reply:
To prevent a circular dependency on imports (it seems on `PolicyDocument`) as done [here](https://github.com/aws/aws-cdk/blob/9d8b06f6478a98e01e3aaa86c1dbf22d6e861f05/packages/aws-cdk-lib/aws-iam/lib/principals.ts#L368).
I added a comment to clarify.

Reply:
Perfect thanks. My next comment was going to be that if I'm confused, so will people who look at the code in the future!